espaloma/graphs/deploy.py

```diff
@@ -0,0 +1,194 @@
+        for position, idxs in enumerate(g.nodes['n3'].data['idxs'])
+    }
+
+    # torsion_lookup = {
+    #     tuple(idxs.detach().numpy()): position
+    #     for position, idxs in enumerate(g.nodes['n4'].data['idxs'])
+    # }
+
+    # create openmm system
+    sys = ff.create_openmm_system(g.mol.to_topology())
+
+    for force in sys.getForces(): # loop through the forces
+        name = force.__class__.__name__ # get the name of the force
+        if 'HarmonicBondForce' in name: # bonds
+            for idx in range(force.getNumBonds()):
+                idx0, idx1, eq, k = force.getBondParameters(idx)
+                position = bond_lookup[(idx0, idx1)]
+                _eq = g.nodes['n2'].data['eq%s' % suffix][position].detach(
+                    ).numpy().item()
+                _k = g.nodes['n2'].data['k%s' % suffix][position].detach(
+                    ).numpy().item()
+
+                _eq = Quantity( # bond length
+                    _eq,
+                    esp.units.DISTANCE_UNIT,
+                ).value_in_unit(OPENMM_BOND_EQ_UNIT)
+
+                _k = 2.0 * Quantity( # bond force constant:
+                    # since everything is enumerated twice in espaloma
+                    # and once in OpenMM,
+                    # we insert a coefficient of 2.0
+                    _k,
+                    esp.units.FORCE_CONSTANT_UNIT,
+                ).value_in_unit(OPENMM_BOND_K_UNIT)
+
+                force.setBondParameters(idx, idx0, idx1, _eq, _k)
+
+        if 'HarmonicAngleForce' in name: # angles
+            for idx in range(force.getNumAngles()):
+                idx0, idx1, idx2, eq, k = force.getAngleParameters(idx)
+                position = angle_lookup[(idx0, idx1, idx2)]
+                _eq = g.nodes['n3'].data['eq%s' % suffix][position].detach(
+                    ).numpy().item()
+                _k = g.nodes['n3'].data['k%s' % suffix][position].detach(
+                    ).numpy().item()
+
+                _eq = Quantity(
+                    _eq,
+                    esp.units.ANGLE_UNIT,
+                ).value_in_unit(OPENMM_ANGLE_EQ_UNIT)
+
+                _k = 2.0 * Quantity( # force constant
+                    # since everything is enumerated twice in espaloma
+                    # and once in OpenMM,
+                    # we insert a coefficient of 2.0
+                    _k,
+                    esp.units.ANGLE_FORCE_CONSTANT_UNIT,
+                ).value_in_unit(OPENMM_ANGLE_K_UNIT)
+
+                force.setAngleParameters(idx, idx0, idx1, idx2, _eq, _k)
+
+        if 'PeriodicTorsionForce' in name:
+            number_of_torsions = force.getNumTorsions()
+            blank_periodicity = False
+            if 'periodicity%s' % suffix not in g.nodes['n4'].data\
+                or 'phase%s' % suffix not in g.nodes['n4'].data:
+                blank_periodicity = True
+
+                g.nodes['n4'].data['periodicity%s' % suffix] = torch.arange(
+                    1, 6
```

Comment:
arange(1, 6) will be 1,2,3,3,4,5 -- should this be arange(1, 7) ?

Reply:
resolved in https://github.com/choderalab/espaloma/pull/39/commits/ee4bc51fb3487c42a1119f6cfa919d5155168c04